lib/uksignal/signal.c

```diff
@@ -297,20 +339,44 @@ int sigwait(const sigset_t *set, int *sig)
 	ptr->wait.status = UK_SIG_NOT_WAITING;
 
 	/* do not execute handler, set received signal */
-	*sig = ptr->wait.received_signal.si_signo;
+	*info = ptr->wait.received_signal;
 
 	/* execute other pending signals */
 	uk_sig_handle_signals();
 
 	return 0; /* returns positive errno */
 }
 
+#if UK_LIBC_SYSCALLS
+int sigwait(const sigset_t *set, int *sig)
+{
+	int error;
+
```

Comment:
What `si`? It does not seem defined to me.
```suggestion
        siginfo_t si;
```